src/tribler/core/components/libtorrent/download_manager/download.py

```diff
@@ -872,6 +883,21 @@ def get_file_index(self, path: Path) -> int:
                     else IllegalFileIndex.expanded_dir.value)
         return IllegalFileIndex.unloaded.value
 
+    @check_handle(None)
+    def set_selected_file_or_dir(self, path: Path, selected: bool) -> None:
+        """
+        Set a single file or directory to be selected or not.
+        """
+        tree = self.tdef.torrent_file_tree
+        prio = 4 if selected else 0
```

Comment:
NIT: the name `prio` is ambiguous. I would suggest renaming it to full `priority`.